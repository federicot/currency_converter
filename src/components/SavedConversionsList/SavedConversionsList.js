import React from 'react';
import { View, StyleSheet } from 'react-native';
import { 
  Icon,
  List,
  ListItem,
  Left,
  Content,
  Text
} from 'native-base';
import PropTypes from 'prop-types';

const propTypes = {
  listItems: PropTypes.arrayOf(PropTypes.shape({
    currencyFrom: PropTypes.string,
    currencyTo: PropTypes.string,
    valueFrom: PropTypes.string,
    valueTo: PropTypes.string,
  })).isRequired,
  onPress: PropTypes.func
};

const defaultProps = {
  onPress: () => {}
}

const styles = StyleSheet.create({
  noConversionsView: {
    flex: 1,
    alignItems: 'center',
    padding: 30,
  }
});

const SavedConversionsList = ({ listItems, onPress }) => {
  return (
    <View>
    {
      listItems.length ? (
        <List>
          {
            [...listItems].reverse().map((conversion, i) => {
              return (
                <ListItem key={i} onPress={onPress(conversion)}>
                  <Left>
                    <Text>{`${conversion.currencyFrom} $${conversion.valueFrom}`}</Text>
                  </Left>
                  <Content>
                    <Icon name='arrow-round-forward' />
                  </Content>
                  <Content>
                    <Text>{`${conversion.currencyTo} $${conversion.valueTo}`}</Text>
                  </Content>
                </ListItem>
              );
            })
          }
        </List>
      ) : (
        <View style={styles.noConversionsView}>
          <Text>You have no saved conversions</Text>
        </View>
      )
    }
    </View>
  );
}

SavedConversionsList.propTypes = propTypes;
SavedConversionsList.defaultProps = defaultProps;

export default SavedConversionsList;