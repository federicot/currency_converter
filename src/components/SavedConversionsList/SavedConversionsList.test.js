/* eslint-env jest */
import React from 'react';
import SavedConversionsList from './SavedConversionsList';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const listItems = [
    {
      currencyFrom: "EUR",
      currencyTo: "USD",
      valueFrom: "10",
      valueTo: "12.31",
    }
  ];
  const tree = renderer.create(
    <SavedConversionsList
      listItems={listItems}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});