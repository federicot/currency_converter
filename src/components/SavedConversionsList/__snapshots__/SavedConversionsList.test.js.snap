// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    closeOnRowBeginSwipe={false}
    closeOnRowPress={true}
    closeOnScroll={true}
    directionalDistanceChangeThreshold={2}
    disableLeftSwipe={false}
    disableRightSwipe={false}
    leftOpenValue={0}
    previewFirstRow={false}
    recalculateHiddenLayout={false}
    rightOpenValue={0}
    style={Object {}}
    swipeToOpenPercent={50}
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        onPress={undefined}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flexDirection": "row",
            "marginLeft": 16,
            "paddingBottom": 13,
            "paddingRight": 16,
            "paddingTop": 13,
          }
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            uppercase={false}
            virtual={undefined}
          >
            EUR $10
          </Text>
        </View>
        <RCTScrollView
          automaticallyAdjustContentInsets={false}
          contentContainerStyle={
            Array [
              Object {
                "padding": undefined,
              },
              undefined,
            ]
          }
          contentInset={
            Object {
              "bottom": 0,
            }
          }
          enableAutomaticScroll={true}
          enableResetScrollToCoords={true}
          extraHeight={75}
          extraScrollHeight={0}
          getScrollResponder={[Function]}
          handleOnScroll={[Function]}
          keyboardDismissMode="interactive"
          keyboardOpeningTime={250}
          keyboardShouldPersistTaps="handled"
          keyboardSpace={0}
          onScroll={[Function]}
          resetKeyboardSpace={[Function]}
          resetScrollToCoords={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          scrollEventThrottle={1}
          scrollForExtraHeightOnAndroid={[Function]}
          scrollToEnd={[Function]}
          scrollToFocusedInput={[Function]}
          scrollToPosition={[Function]}
          showsVerticalScrollIndicator={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
            }
          }
          viewIsInsideTabBar={false}
          virtual={undefined}
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
              virtual={undefined}
            >
              ï„™
            </Text>
          </View>
        </RCTScrollView>
        <RCTScrollView
          automaticallyAdjustContentInsets={false}
          contentContainerStyle={
            Array [
              Object {
                "padding": undefined,
              },
              undefined,
            ]
          }
          contentInset={
            Object {
              "bottom": 0,
            }
          }
          enableAutomaticScroll={true}
          enableResetScrollToCoords={true}
          extraHeight={75}
          extraScrollHeight={0}
          getScrollResponder={[Function]}
          handleOnScroll={[Function]}
          keyboardDismissMode="interactive"
          keyboardOpeningTime={250}
          keyboardShouldPersistTaps="handled"
          keyboardSpace={0}
          onScroll={[Function]}
          resetKeyboardSpace={[Function]}
          resetScrollToCoords={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          scrollEventThrottle={1}
          scrollForExtraHeightOnAndroid={[Function]}
          scrollToEnd={[Function]}
          scrollToFocusedInput={[Function]}
          scrollToPosition={[Function]}
          showsVerticalScrollIndicator={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
            }
          }
          viewIsInsideTabBar={false}
          virtual={undefined}
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
              virtual={undefined}
            >
              USD $12.31
            </Text>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</View>
`;
